legacy/plugin.ts

```diff
@@ -92,15 +92,17 @@ export interface PluginMetadata {
   // Per-plugin custom metadata
   meta?: {
     allSubProjectNames?: string[],
-    versionBuildInfo?: VersionBuildInfo,
+    gradleVersionBuildInfo?: GradleVersionBuildInfo,
+    javaVersion?: string,
+    mavenVersion?: string,
   };
 
   // Docker-related fields
   dockerImageId?: any;
   imageLayers?: any;
 }
 
-export interface VersionBuildInfo {
+export interface GradleVersionBuildInfo {
   gradleVersion: string;
   metaBuildVersion: { [index: string]: string };
```

Comment:
Was it intended that this `metaBuildVersion` could cater for all version metadata?
So for your purposes:

```
const versionBuildInfo: VersionBuildInfo = {
  metaBuildVersion: {
    "java": "X.X.X",
    "maven": "X.X.X",
  }
}
```

In which case I think I prefer the more generic original, and to remove `gradleVersion: string;` incorporating it into `metaBuildVersion` as "gradle" key?

Reply:
I can happily remove `gradleVersion`, I just took another approach :-)

Reply:
yep think it makes sense not to list all languages/build tools as separate properties, as we probably will want to report versions on all sorts eventually.

Reply:
Reverting back and making `gradleVersion` optional to allow non-gradle plugins.